### PR TITLE
Added unsupervised method. Changed to best hparams.

### PR DESCRIPTION
**Pull request by wichersn**

### CODE REVIEW

**Comment by MislavJuric** [comment]
Code review sounds like a good idea (maybe on the weekly meeting)? I "got the feel" of the code, but feel I could use someone stepping through it.

**Comment by ricvolpe** [comment]
Maybe @arunraja-hub can install [ReviewNB (Github App)](https://www.reviewnb.com/) so that we can review notebooks more easily?

**Comment by ricvolpe** [comment]
I tagged Arun because I believe only he can install GitHub apps that become available in this repo. I tried installing it but I see only my repositories at https://app.reviewnb.com/

File: extract_preferences.ipynb [hunk 1]
```diff
@@ -342,7 +480,7 @@
         "                        ],\n",
         "                )\n",
         "  return model\n",
-        "all_hparam_possibilities = [{\"reg_amount\": [0.0], \"drop_rate\": [0.0],}]"
+        "all_hparam_possibilities = [{\"reg_amount\": [0.0], \"drop_rate\": [0.0], 'reduction_alg': [None], 'n_components': [None]}]"
```
Comment (ricvolpe) [inline extract_preferences.ipynb]: Isn't this line now overrode by later cells (with/without unsupervised dimensionality reduction)?

**Comment by ricvolpe** [comment]
Taylor and I took a look during the call. Overall it looks good to us ;)

File: extract_preferences.ipynb (hunk 1 above)
Comment (wichersn) [reply extract_preferences.ipynb]: Clarified this in the comments for the cells below.

**Pull request merged**